packages/vue3-view360/src/View360.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2023-present NAVER Corp.
+ * egjs projects are licensed under the MIT license
+ */
+import { defineComponent, PropType, h } from "vue";
+import VanillaView360, {
+  DEFAULT_CLASS,
+  EVENTS,
+  getValidProps,
+  withMethods
+} from "@egjs/view360";
+
+import { View360Props } from "./types";
+
+type PropTypes = {
+  [key in keyof View360Props]: {
+    type: PropType<View360Props[key]>;
+  }
+};
+
+const view360OptionNames = Object.getOwnPropertyNames(VanillaView360.prototype)
+  .filter(name => {
+    const descriptor = Object.getOwnPropertyDescriptor(VanillaView360.prototype, name);
+
+    if (name.startsWith("_")) return false;
+    if (descriptor?.value) return false;
+
+    return true;
+  });
+
+const view360Options = view360OptionNames.reduce((props, name) => {
+    return {
+      ...props,
+      [name]: null
+    };
+  }, {}) as PropTypes;
+
+const view360SetterNames = view360OptionNames.filter(name => {
+  const descriptor = Object.getOwnPropertyDescriptor(VanillaView360.prototype, name);
+
+  return !!descriptor!.set;
+});
+
+const View360 = defineComponent({
+  props: {
+    ...view360Options,
+    tag: {
+      type: String,
+      required: false,
+      default: "div"
+    },
+    canvasClass: {
+      type: String,
+      required: false,
+      default: ""
+    }
+  },
+  data() {
+    return {} as {
+      view360: VanillaView360;
+    };
+  },
+  created() {
+    withMethods(this, "view360");
```

Comment:
If you declare a method like this, a warning occurs.

It must be properly declared through data, methods, or setup.